extensions/arc/src/models/resourceModel.ts

```diff
@@ -23,5 +26,61 @@ export abstract class ResourceModel {
 		this._onRegistrationUpdated.fire(this._registration);
 	}
 
+	/**
+ * Loads the saved connection profile associated with this model. Will prompt for one if
+ * we don't have one or can't find it (it was deleted)
+ */
+	protected async getConnectionProfile(externalEndpoint: string | undefined, providerName: string, userName: string | undefined): Promise<void> {
+		const ipAndPort = parseIpAndPort(externalEndpoint || '');
+		let connectionProfile: azdata.IConnectionProfile | undefined = {
+			serverName: `${ipAndPort.ip},${ipAndPort.port}`,
```

Comment:
Is this going to work for PG connections? In my testing when I had the `,port` in the server name field it failed

Reply:
Having the host and port defined in the options avoids using the servername when making the connection so no failure. (Left servername defined for PG so that the connectSQLDialog can still grab it to show to user).